internal/directory/google/google.go

```diff
@@ -82,62 +82,62 @@ func New(options ...Option) *Provider {
 // NOTE: groups via Directory API is limited to 1 QPS!
 // https://developers.google.com/admin-sdk/directory/v1/reference/groups/list
 // https://developers.google.com/admin-sdk/directory/v1/limits
-func (p *Provider) UserGroups(ctx context.Context) ([]*directory.User, error) {
+func (p *Provider) UserGroups(ctx context.Context) ([]*directory.Group, []*directory.User, error) {
 	apiClient, err := p.getAPIClient(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("google: error getting API client: %w", err)
+		return nil, nil, fmt.Errorf("google: error getting API client: %w", err)
 	}
 
-	groupIDToEmails := map[string]string{}
-	var groups []string
+	var groups []*directory.Group
 	err = apiClient.Groups.List().
 		Context(ctx).
-		Fields("id", "email", "directMembersCount").
 		Customer("my_customer").
 		Pages(ctx, func(res *admin.Groups) error {
 			for _, g := range res.Groups {
 				// Skip group without member.
 				if g.DirectMembersCount == 0 {
 					continue
 				}
-				groups = append(groups, g.Id)
-				groupIDToEmails[g.Id] = g.Email
+				groups = append(groups, &directory.Group{
+					Id:    g.Id,
+					Name:  g.Name,
```

Comment:
To be backwards compatible, I think we should use Group as Name. 

https://github.com/pomerium/pomerium/blob/50f272fa707d77941ea723dea02715ea8db9d777/internal/identity/oidc/google/google.go#L121-L123

Reply:
Not sure I understand the reasoning here, but I will change it.